text/0000-new-test-waiters.md

```diff
@@ -85,21 +85,114 @@ export default class Friendz extends Component {
 
 In the above example, a new test waiter is built that is identified via a `name` string passed into the `buildWaiter` function. This allows the waiter to be identifiable, and that name is ultimately used with test isolation validation to help developers narrow down problems in their tests.
 
-The API used to signal whether an asynchronous operation has begun and ultimately ended is through the paired calls of `beginAsync` and `endAsync`: begin to denote the start of the asynchronous operation, and end to denote the end. Unique instances of async operations are identified using a `token` returned from `beginAsync`, which is subsequently provided to the `endAsync` call.
+## New Test Waiters Design
+
+The new test waiters addon is built using low-level primitives that are complimented with some convenience utilities.
+
+### `IWaiter`
+
+At its core, the addon uses an `IWaiter` interface defined as follows:
+
+```ts
+export type WaiterName = string;
+export type Token = unknown;
+
+export interface IWaiter {
+  name: WaiterName;
+  waitUntil(): boolean;
+  debugInfo(): ITestWaiterDebugInfo[];
+}
+```
+
+This allows for maximum flexibility when creating your own waiter implementations.
+
+### `ITestWaiter`
+
+The `IWaiter` interface is built upon to create a more specific interface for a test waiter, `ITestWaiter`:
+
+```ts
+export interface ITestWaiter<T = Token> extends IWaiter {
+  beginAsync(token?: T, label?: string): T;
+  endAsync(token: T): void;
+  reset(): void;
+}
+```
+
+This interface is used for the two internal concrete types: `TestWaiter` and `NoopTestWaiter`. These types for the basis for the addon, and will likely satisfy the majority of use cases.
```

Comment:
> These types for the basis

This confuses me a bit, maybe you meant:

```
These types form the basis
```

Reply:
Ya that's a typo.